File: Source/code/forms/formPlaybackSpeed.pas
// ===================================================================
//
// (c) Paul Alan Freshney 2012-2023
// www.freshney.org :: paul@freshney.org :: maximumoctopus.com
//
// https://github.com/MaximumOctopus/LEDMatrixStudio
//
// Please do not modifiy this comment section
//
// ===================================================================

unit formPlaybackSpeed;

interface

uses
  Winapi.Windows, Winapi.Messages, System.SysUtils, System.Variants, System.Classes, Vcl.Graphics,
  Vcl.Controls, Vcl.Forms, Vcl.Dialogs, Vcl.StdCtrls, Vcl.Buttons, Vcl.ExtCtrls,

  languagehandler;

type
  TfrmCustomPlayback = class(TForm)
    GroupBox1: TGroupBox;
    bOK: TBitBtn;
    bCancel: TBitBtn;
    eSpeed: TEdit;
    Label1: TLabel;
    Image1: TImage;
    Bevel1: TBevel;
    lEquality: TLabel;
    procedure bOKClick(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    procedure SetGUILanguageText;
  public
  end;


var
  frmCustomPlayback: TfrmCustomPlayback;


function DoCustomPlaybackSpeed(aOldCustom : integer): integer;


implementation


{$R *.dfm}


function DoCustomPlaybackSpeed(aOldCustom : integer): integer;
begin
  with TfrmCustomPlayback.Create(Application) do
    try
      eSpeed.Text := IntToStr(aOldCustom);

      ShowModal;

      if ModalResult = mrOK then
        Result := StrToIntDef(eSpeed.Text, 0)
      else
        Result := 0;
    finally
      Free;
    end;
end;


procedure TfrmCustomPlayback.FormCreate(Sender: TObject);
begin
  SetGUILanguageText;
end;


procedure TfrmCustomPlayback.SetGUILanguageText;
begin
  Caption := GLanguageHandler.Text[kCustomPlaybackSpeed];
  lEquality.Caption := GLanguageHandler.Text[k1000ms1Second];

  bOK.Caption := GLanguageHandler.Text[kOK];
  bCancel.Caption := GLanguageHandler.Text[kCancel];
end;


procedure TfrmCustomPlayback.bOKClick(Sender: TObject);
var
  lSpeed : integer;

begin
  lSpeed := StrToIntDef(eSpeed.Text, 0);

  if (lSpeed > 0) then
    ModalResult := mrOK;
end;


end.

File: Source/Forms/FormPlaybackSpeed.cpp
// ===================================================================
//
//   (c) Paul Alan Freshney 2012-2024
//   www.freshney.org :: paul@freshney.org :: maximumoctopus.com
//
//   https://github.com/MaximumOctopus/LEDMatrixStudio
//
//   https://maximumoctopus.hashnode.dev/
//
//   C++ Rewrite October 11th 2023
//
// ===================================================================

#include <vcl.h>
#pragma hdrstop

#include "FormPlaybackSpeed.h"
#include "LanguageConstants.h"
#include "LanguageHandler.h"

#pragma package(smart_init)
#pragma resource "*.dfm"
TfrmPlaybackSpeed *frmPlaybackSpeed;

extern LanguageHandler *GLanguageHandler;


//---------------------------------------------------------------------------
//---------------------------------------------------------------------------

int OpenCustomPlaybackSpeed(int oldspeed)
{
	TfrmPlaybackSpeed *frmPlaybackSpeed = new TfrmPlaybackSpeed(Application);

	int speed = oldspeed;

	frmPlaybackSpeed->eSpeed->Text = speed;

	if (frmPlaybackSpeed->ShowModal() == mrOk)
	{
		speed = frmPlaybackSpeed->eSpeed->Text.ToIntDef(0);
	}

	delete frmPlaybackSpeed;

	return speed;
}

//---------------------------------------------------------------------------
//---------------------------------------------------------------------------


__fastcall TfrmPlaybackSpeed::TfrmPlaybackSpeed(TComponent* Owner)
	: TForm(Owner)
{
	SetGuiLanguageText();
}


void __fastcall TfrmPlaybackSpeed::bOKClick(TObject *Sender)
{
	int speed = eSpeed->Text.ToIntDef(0);

	if (speed > 0)
	{
		ModalResult = mrOk;
	}
}


void TfrmPlaybackSpeed::SetGuiLanguageText()
{
	Caption = GLanguageHandler->Text[kCustomPlaybackSpeed].c_str();
	lEquality->Caption = GLanguageHandler->Text[k1000ms1Second].c_str();

	bOK->Caption = GLanguageHandler->Text[kOK].c_str();
	bCancel->Caption = GLanguageHandler->Text[kCancel].c_str();
}
